apps/src/storage/firebaseStorage.js

```diff
@@ -260,25 +260,34 @@ FirebaseStorage.readRecords = function(
 ) {
   tableName = fixTableName(tableName, onError);
 
-  let recordsRef = getRecordsRef(tableName);
-
-  // Get all records in the table and filter them on the client.
-  recordsRef.once(
-    'value',
-    recordsSnapshot => {
-      let recordMap = recordsSnapshot.val() || {};
-      let records = [];
-      // Collect all of the records matching the searchParams.
-      Object.keys(recordMap).forEach(id => {
-        let record = JSON.parse(recordMap[id]);
-        if (matchesSearch(record, searchParams)) {
-          records.push(record);
-        }
-      });
-      onSuccess(records);
-    },
-    onError
-  );
+  const countersRef = getDatabase().child('counters/tables/');
+  // First, check if table exists using counters/tables/ as source of truth
+  countersRef.once('value', countersSnapshot => {
+    if (!countersSnapshot.val() || !countersSnapshot.val()[tableName]) {
```

Comment:
Is it possible for someone to name a table `0` and break this condition?

Reply:
ha, Good question! I wasn't sure so I checked and looks like it works as expected: 
![image](https://user-images.githubusercontent.com/8787187/62667087-fc153200-b93a-11e9-9a78-0d1f929ec47a.png)

If there are tables, `countersSnapshot.val()` will be an object with a key for each table, with the value being that table's counters. In this case, it's `{0: {lastId: 1, rowCount: 0}}`, which is truthy. Currently, the only possible way it would be falsy is if there are no tables, in which case `countersSnapshot.val()` is `null`.